tests/unittests/unittests.cpp

```diff
@@ -538,6 +557,58 @@ class ReaderWriterQueueTests : public TestClass<ReaderWriterQueueTests>
 
 		return true;
 	}
+
+#if MOODYCAMEL_HAS_EMPLACE
+	bool emplace()
+	{
+		ReaderWriterQueue<UniquePtrWrapper> q(100);
+		std::unique_ptr<int> p { new int(123) };
+		q.emplace(std::move(p));
+		UniquePtrWrapper item;
+		ASSERT_OR_FAIL(q.try_dequeue(item));
+		ASSERT_OR_FAIL(item.get_value() == 123);
+		ASSERT_OR_FAIL(q.size_approx() == 0);
+
+		return true;
+	}
+
+	// This is what you have to do to try_enqueue() a movable type, and demonstrates why try_emplace() is useful
+	bool try_enqueue_bad()
```

Comment:
Ah, I understand now. Thanks :-)
Maybe rename this to `try_emplace_fail_workaround`?

Reply:
Done.

Reply:
Actually try_enqueue_fail_workaround() surely?

Reply:
Mmm could go either way :D
I meant as a workaround for the method `try_emplace_fail` not existing, but it uses `try_enqueue` to accomplish that so... up to you!